TESTING.md

```diff
@@ -0,0 +1,99 @@
+# tardigrade-ci testing documentation
+
+Testing tardigrade-ci implementation with various setup on local environment.
+
+## Testing
+
+This project can be utilized one of two ways, via docker or via a Makefile include.
+
+### Via Docker (non-tardigrade-ci image)
+
+In a non-tardigrade-ci container, the second condition in Makefile.bootstrap ought to fail, resulting in the auto-init logic creating the tardigrade-ci/ directory.
+
+1. Run the following bash script or manually create a test-project directory with the following configurations:
+
+```bash
+#!/bin/bash
+
+mkdir test-project
```

Comment:
I'd suggest setting up in TMPDIR for this testing, and using `mktemp` to create the test directory.

```suggestion
TESTDIR=$(mktemp --tmpdir -d tardigrade-ci.XXXXXXXXXX)
```

and then references to `test-project` should be changed to `"$TESTDIR"`